services/appRoutes.mjs

```diff
@@ -0,0 +1,41 @@
+import express from "express";
+import fetch from "node-fetch";
+
+const appRoutes = express.Router();
+
+
+// Route to render a list of plants
+appRoutes.get("/", async (req, res) => {
+  try {
+    const response = await fetch(`http://localhost:3005/api/plants`);
+
+    const data = await response.json();
+    console.log("in app routes", data);
+
+    res.render("index.njk", {
+      title: "Plants List",
+      page: "plants",
+      plants: data.data,
+    });
+  } catch (error) {
+    console.error("Error fetching plants:", error);
+    res.status(500).send("Error fetching plants.");
+  }
+});
+
+// Route for a specific plant
+appRoutes.get("/plant/:name", async (req, res) => {
+  const { name } = req.params;
+  try {
+    console.log("in plants", name);
+
+    const response = await fetch(`http://localhost:3005/api/plant/${name}`);
```

Comment:
This approach would force the user to wait for 3 HTTP requests, at most we want 2. The parts in `apiRoutes` that make the calls should be in a service. Something like:

```js
// service/plants.js

const handleRequest = async (path, params = {}) = {
  const response = await fetch(`${API_URL}/${path}`);
  // handle response

  return response.json();
}

const all = async () => {
  const results = await handleRequest('/plants');

  return results.map(result => ({ /* some subset of fields */ }));
}

const get = async (plantId) => {
  const result = await handleRequest(`/plants/${id}`);

  return { /* some subset of fields */ }
}

/* snip */
```